ingest/ledgerbackend/rpc_backend.go

```diff
@@ -0,0 +1,293 @@
+package ledgerbackend
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"net/http"
+	"sync"
+	"time"
+
+	"github.com/stellar/go/xdr"
+	rpc "github.com/stellar/stellar-rpc/client"
+	"github.com/stellar/stellar-rpc/protocol"
+)
+
+const rpcBackendDefaultBufferSize uint32 = 10
+const rpcBackendDefaultWaitIntervalSeconds uint32 = 2
+
+type RPCLedgerMissingError struct {
+	Sequence uint32
+}
+
+func (e *RPCLedgerMissingError) Error() string {
+	return fmt.Sprintf("ledger %d was not present on rpc", e.Sequence)
+}
+
+type RPCLedgerBeyondLatestError struct {
+	Sequence     uint32
+	LatestLedger uint32
+}
+
+func (e *RPCLedgerBeyondLatestError) Error() string {
+	return fmt.Sprintf("ledger %d is beyond the RPC latest ledger is %d", e.Sequence, e.LatestLedger)
+}
+
+// The minimum required RPC client interface required for usage by RPCLedgerBackend.
+type RPCClient interface {
+	GetLedgers(ctx context.Context, req protocol.GetLedgersRequest) (protocol.GetLedgersResponse, error)
+}
+
+type RPCLedgerBackend struct {
+	client        RPCClient
+	buffer        map[uint32]xdr.LedgerCloseMeta
+	bufferSize    uint32
```

Comment:
```suggestion
	limit    uint32
```

the name `bufferSize` makes it seem that it would be equal to `len(buffer)`